File: C/Cap07/c07ex05.c
// c07ex05.c

#include <stdio.h>

void FATORIAL(int N, int *Fat) {
  *Fat = 1;
  for (int I = 1; I <= N; I++) {
    *Fat *= I;
  }
}

int Resp, Limite;
  
int main(void) {
  Resp = 1;

  printf("Qual fatorial: ");
  scanf("%d", &Limite);
  while (getchar() != '\n');

  FATORIAL(Limite, &Resp);

  printf("Fatorial = %d\n", Resp);

  printf("\n");
  printf("Tecle <Enter> para encerrar... ");
  while (getchar() != '\n');

  return 0;
}

File: TYPESCRIPT/Cap07/c07ex05.ts
// c07ex05.ts

import * as rls from 'readline-sync';

function FATORIAL(N: number, Fat: number[]): void {
  Fat[0] = 1;
  for (let I: number = 1; I <= N; I++) {
    Fat[0] *= I;
  }
}

let Resp: number[], Limite: number;

Limite = rls.questionInt("Qual fatorial: ");
Resp = [0];

FATORIAL(Limite, Resp);

console.log(`Fatorial = ${Resp[0]}`);

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();
